tomopy/prep/stripe.py

```diff
@@ -337,3 +348,450 @@ def remove_stripe_sf(tomo, size=5, ncore=None, nchunk=None):
+    matsortback = np.asarray(
+        [row[row[:, 0].argsort()] for row in matsort])
+    sino_corrected = np.transpose(matsortback[:, :, 1])
+    listxmiss = np.where(listmask > 0.0)[0]
+    sinogram[:, listxmiss] = sino_corrected[:, listxmiss]
+    return sinogram
+
+
+def _remove_large_stripe(tomo, snr, size):
+    matindex = _create_matindex(tomo.shape[2], tomo.shape[0])
+    for m in range(tomo.shape[1]):
+        sino = tomo[:, m, :]       
+        tomo[:, m, :] = _rs_algorithm_5(sino, snr, size, matindex)
+
+
+def remove_dead_stripe(tomo, snr=3, size=51, ncore=None, nchunk=None):
+    """
+    Remove horizontal stripes from sinogram using Nghia Vo's
+    approach :cite:
+    Algorithm 6 in the paper. Remove unresponsive and fluctuating stripes.
+    Parameters
+    ----------
+    tomo : ndarray
+        3D tomographic data.
+    snr  : float
+        Ratio used to detect locations of large stripes.
+        Greater is less sensitive. 
+    size : int
+        Window size of the median filter.
+    ncore : int, optional
+        Number of cores that will be assigned to jobs.
+    nchunk : int, optional
+        Chunk size for each core.
+ 
+    Returns
+    -------
+    ndarray
+        Corrected 3D tomographic data.
+    """
+    arr = mproc.distribute_jobs(
+        tomo,
+        func=_remove_dead_stripe,
+        args=(snr, size),
+        axis=1,
+        ncore=ncore,
+        nchunk=nchunk)
+    return arr
+
+
+def _rs_algorithm_6(sinogram, snr, size, matindex):
+    """
+    Remove unresponsive and fluctuating stripes.
+    """
+    sinogram = np.copy(sinogram) # Make it mutable
+    (nrow, _) = sinogram.shape
+    sinosmoothed = np.apply_along_axis(uniform_filter1d, 0, sinogram, 10)
+    listdiff = np.sum(np.abs(sinogram - sinosmoothed), axis=0)
+    nmean = np.mean(listdiff)
+    listdiffbck = median_filter(listdiff, size)
+    listdiffbck[listdiffbck == 0.0] = nmean
+    listfact = listdiff / listdiffbck
+    listmask = _detect_stripe(listfact, snr)
+    listmask = binary_dilation(listmask, iterations=1).astype(listmask.dtype)
+    listmask[0:2] = 0.0
+    listmask[-2:] = 0.0
+    listx = np.where(listmask < 1.0)[0]
+    listy = np.arange(nrow)
+    matz = sinogram[:, listx]
+    finter = interpolate.interp2d(listx, listy, matz, kind='linear')
+    listxmiss = np.where(listmask > 0.0)[0]
+    if len(listxmiss) > 0:
+        matzmiss = finter(listxmiss, listy)
+        sinogram[:, listxmiss] = matzmiss
+    # Use algorithm 5 to remove residual stripes
+    sinogram = _rs_algorithm_5(sinogram, snr, size, matindex) 
+    return sinogram
+
+def _remove_dead_stripe(tomo, snr, size):
+    matindex = _create_matindex(tomo.shape[2], tomo.shape[0])
+    for m in range(tomo.shape[1]):
+        sino = tomo[:, m, :]       
+        tomo[:, m, :] = _rs_algorithm_6(sino, snr, size, matindex)
+        
+
+def remove_all_stripe(tomo, snr=3, la_size=61, sm_size=21, ncore=None, nchunk=None):
+    """
```

Comment:
Can snr be calculated from data?

Reply:
snr is a threshold used to segment stripes (name of snr may cause confusion).  Users have to specify it.